### PR TITLE
* [tosa] Support for AtenFlattenUsingIntsOp

### PR DESCRIPTION
**Pull request by anupgangwar**
Signed-off-by: Anup Gangwar <anup.gangwar@arm.com>

### CODE REVIEW

**Comment by anupgangwar** [comment]
Summary:
    Passed: 65
    Expectedly Failed: 187

Tagging @cathyzhyi for review.

**Comment by sjarus** [comment]
Thank you for these fixes @ljfitz and @cathyzhyi! I'm looking forward to trying this out with my maxpool/adaptive avgpool TOSA implementations once it's merged.

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp [hunk 1]
```diff
@@ -1928,6 +1928,61 @@ LogicalResult ConvertAtenOp<ValueTensorLiteralOp>::matchAndRewrite(
...
+  if (!matchPattern(op.end_dim(), m_TorchConstantInt(&end_dim)))
+    return op.emitError("end_dim must be a Scalar constant");
+
+  if (end_dim != -1 && end_dim < start_dim)
+    return op.emitError("end_dim must be larger than start_dim");
+
+  SmallVector<int64_t> newShape;
+  uint32_t idx = 0;
+  bool acc = false;
+  for (auto s : selfType.getShape()) {
```
Comment (cathyzhyi) [inline lib/Conversion/TorchToTosa/TorchToTosa.cpp:1956]: nit: could use llvm::enumerate() for this, then the var `idx` would not be needed. see example here https://github.com/llvm/mlir-npcomp/blob/52ed3313b49b8f56a6dac270ee334c6dffebcaf2/lib/Conversion/TorchToLinalg/TorchToLinalg.cpp#L2886

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp [hunk 2]
```diff
@@ -1928,6 +1928,61 @@ LogicalResult ConvertAtenOp<ValueTensorLiteralOp>::matchAndRewrite(
...
+    ++idx;
+  }
+
+  // Handle the Scalar case
+  if (newShape.size() == 0)
+    newShape.push_back(1);
+
+  auto newType = RankedTensorType::get(newShape, selfType.getElementType());
+  auto reshapeOp = rewriter.create<tosa::ReshapeOp>(
+      op->getLoc(), getTypeConverter()->convertType(newType), adaptor.self(),
```
Comment (cathyzhyi) [inline lib/Conversion/TorchToTosa/TorchToTosa.cpp:1977]: newType is already builtin tensor type and the type conversion is not needed.

**Review by cathyzhyi — CHANGES_REQUESTED** [review]

**Review by cathyzhyi — APPROVED** [review]

**Pull request merged**